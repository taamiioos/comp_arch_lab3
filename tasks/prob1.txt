.data:
    sum NUMBER 0

.code:
    MOV rc, 0

loop:
    CMP rc, 1000
    JZ end_loop
    MOD r2, r1, 3
    CMP r2, 0
    JZ add_to_sum
    MOD r2, r1, 5
    CMP r2, 0
    JNZ next_iter

add_to_sum:
    ADD sum, r1

next_iter:
    INC rc
    JMP loop

end_loop:
    LOAD r1, sum
    OUTPUT r1
    HLT

