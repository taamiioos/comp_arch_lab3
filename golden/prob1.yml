in_source: |-
  .data:
      NUMBER sum 0

  .code:
      MOV r1 1

  loop:
      CMP r1 1000
      JZ end_loop
      MOD_RRV r2 r1 3
      JZ_R add_to_sum r2
      MOD_RRV r2 r1 5
      JZ_R add_to_sum r2
      JMP next_iter

  add_to_sum:
      ADD_MMR sum sum r1

  next_iter:
      INC r1
      JMP loop

  end_loop:
      LOAD_V r2 sum
      OUTPUT r2
      HLT

in_stdin: |
out_stdout: |
  LoC: 26 Instr: 18
  ============================================================
  233168
  Instructions: 8331 Ticks: 13060
out_code: |-
  [3,
   {"index": 0, "opcode": "number", "args": [1234]},
   {"index": 1, "opcode": "number", "args": [1234]},
   {"index": 2, "opcode": "number", "args": [0]},
   {"index": 0, "opcode": "mov", "args": [0, 1]},
   {"index": 1, "opcode": "cmp", "args": [0, 1000]},
   {"index": 2, "opcode": "jz", "args": [11]},
   {"index": 3, "opcode": "mod_rrv", "args": [1, 0, 3]},
   {"index": 4, "opcode": "jz_r", "args": [8, 1]},
   {"index": 5, "opcode": "mod_rrv", "args": [1, 0, 5]},
   {"index": 6, "opcode": "jz_r", "args": [8, 1]},
   {"index": 7, "opcode": "jmp", "args": [9]},
   {"index": 8, "opcode": "add_mmr", "args": [2, 2, 0]},
   {"index": 9, "opcode": "inc", "args": [0]},
   {"index": 10, "opcode": "jmp", "args": [1]},
   {"index": 11, "opcode": "load_v", "args": [1, 2]},
   {"index": 12, "opcode": "store_v", "args": [1, 1]},
   {"index": 13, "opcode": "hlt", "args": []}]
out_log: |
  INFO     root:simulation.py:60 Start simulation
  DEBUG    root:simulation.py:26 TICK:    1 [ 1: CMP    ] PC:  1 DR:  0 SB:  0 RG: [1, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:    2 [ 2: JZ     ] PC:  2 DR:  0 SB:  0 RG: [1, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:    3 [ 3: MOD_RRV] PC:  3 DR:  0 SB:  0 RG: [1, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:    5 [ 4: JZ_R   ] PC:  4 DR:  0 SB:  0 RG: [1, 1, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:    7 [ 5: MOD_RRV] PC:  5 DR:  0 SB:  0 RG: [1, 1, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:    9 [ 6: JZ_R   ] PC:  6 DR:  0 SB:  0 RG: [1, 1, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   11 [ 7: JMP    ] PC:  7 DR:  0 SB:  0 RG: [1, 1, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   12 [ 9: INC    ] PC:  9 DR:  0 SB:  0 RG: [1, 1, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   13 [10: JMP    ] PC: 10 DR:  0 SB:  0 RG: [2, 1, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   14 [ 1: CMP    ] PC:  1 DR:  0 SB:  0 RG: [2, 1, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   15 [ 2: JZ     ] PC:  2 DR:  0 SB:  0 RG: [2, 1, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   16 [ 3: MOD_RRV] PC:  3 DR:  0 SB:  0 RG: [2, 1, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   18 [ 4: JZ_R   ] PC:  4 DR:  0 SB:  0 RG: [2, 2, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   20 [ 5: MOD_RRV] PC:  5 DR:  0 SB:  0 RG: [2, 2, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   22 [ 6: JZ_R   ] PC:  6 DR:  0 SB:  0 RG: [2, 2, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   24 [ 7: JMP    ] PC:  7 DR:  0 SB:  0 RG: [2, 2, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   25 [ 9: INC    ] PC:  9 DR:  0 SB:  0 RG: [2, 2, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   26 [10: JMP    ] PC: 10 DR:  0 SB:  0 RG: [3, 2, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   27 [ 1: CMP    ] PC:  1 DR:  0 SB:  0 RG: [3, 2, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   28 [ 2: JZ     ] PC:  2 DR:  0 SB:  0 RG: [3, 2, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   29 [ 3: MOD_RRV] PC:  3 DR:  0 SB:  0 RG: [3, 2, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   31 [ 4: JZ_R   ] PC:  4 DR:  0 SB:  0 RG: [3, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   33 [ 8: ADD_MMR] PC:  8 DR:  0 SB:  0 RG: [3, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   37 [ 9: INC    ] PC:  9 DR:  2 SB:  0 RG: [3, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   38 [10: JMP    ] PC: 10 DR:  2 SB:  0 RG: [4, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   39 [ 1: CMP    ] PC:  1 DR:  2 SB:  0 RG: [4, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   40 [ 2: JZ     ] PC:  2 DR:  2 SB:  0 RG: [4, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   41 [ 3: MOD_RRV] PC:  3 DR:  2 SB:  0 RG: [4, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   43 [ 4: JZ_R   ] PC:  4 DR:  2 SB:  0 RG: [4, 1, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   45 [ 5: MOD_RRV] PC:  5 DR:  2 SB:  0 RG: [4, 1, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   47 [ 6: JZ_R   ] PC:  6 DR:  2 SB:  0 RG: [4, 4, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   49 [ 7: JMP    ] PC:  7 DR:  2 SB:  0 RG: [4, 4, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   50 [ 9: INC    ] PC:  9 DR:  2 SB:  0 RG: [4, 4, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   51 [10: JMP    ] PC: 10 DR:  2 SB:  0 RG: [5, 4, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   52 [ 1: CMP    ] PC:  1 DR:  2 SB:  0 RG: [5, 4, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   53 [ 2: JZ     ] PC:  2 DR:  2 SB:  0 RG: [5, 4, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   54 [ 3: MOD_RRV] PC:  3 DR:  2 SB:  0 RG: [5, 4, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   56 [ 4: JZ_R   ] PC:  4 DR:  2 SB:  0 RG: [5, 2, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   58 [ 5: MOD_RRV] PC:  5 DR:  2 SB:  0 RG: [5, 2, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   60 [ 6: JZ_R   ] PC:  6 DR:  2 SB:  0 RG: [5, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   62 [ 8: ADD_MMR] PC:  8 DR:  2 SB:  0 RG: [5, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   66 [ 9: INC    ] PC:  9 DR:  2 SB:  3 RG: [5, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   67 [10: JMP    ] PC: 10 DR:  2 SB:  3 RG: [6, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   68 [ 1: CMP    ] PC:  1 DR:  2 SB:  3 RG: [6, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   69 [ 2: JZ     ] PC:  2 DR:  2 SB:  3 RG: [6, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   70 [ 3: MOD_RRV] PC:  3 DR:  2 SB:  3 RG: [6, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   72 [ 4: JZ_R   ] PC:  4 DR:  2 SB:  0 RG: [6, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   74 [ 8: ADD_MMR] PC:  8 DR:  2 SB:  0 RG: [6, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   78 [ 9: INC    ] PC:  9 DR:  2 SB:  8 RG: [6, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   79 [10: JMP    ] PC: 10 DR:  2 SB:  8 RG: [7, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   80 [ 1: CMP    ] PC:  1 DR:  2 SB:  8 RG: [7, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   81 [ 2: JZ     ] PC:  2 DR:  2 SB:  8 RG: [7, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   82 [ 3: MOD_RRV] PC:  3 DR:  2 SB:  8 RG: [7, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   84 [ 4: JZ_R   ] PC:  4 DR:  2 SB:  0 RG: [7, 1, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   86 [ 5: MOD_RRV] PC:  5 DR:  2 SB:  0 RG: [7, 1, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   88 [ 6: JZ_R   ] PC:  6 DR:  2 SB:  0 RG: [7, 2, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   90 [ 7: JMP    ] PC:  7 DR:  2 SB:  0 RG: [7, 2, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   91 [ 9: INC    ] PC:  9 DR:  2 SB:  0 RG: [7, 2, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   92 [10: JMP    ] PC: 10 DR:  2 SB:  0 RG: [8, 2, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   93 [ 1: CMP    ] PC:  1 DR:  2 SB:  0 RG: [8, 2, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   94 [ 2: JZ     ] PC:  2 DR:  2 SB:  0 RG: [8, 2, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   95 [ 3: MOD_RRV] PC:  3 DR:  2 SB:  0 RG: [8, 2, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   97 [ 4: JZ_R   ] PC:  4 DR:  2 SB:  0 RG: [8, 2, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:   99 [ 5: MOD_RRV] PC:  5 DR:  2 SB:  0 RG: [8, 2, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  101 [ 6: JZ_R   ] PC:  6 DR:  2 SB:  0 RG: [8, 3, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  103 [ 7: JMP    ] PC:  7 DR:  2 SB:  0 RG: [8, 3, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  104 [ 9: INC    ] PC:  9 DR:  2 SB:  0 RG: [8, 3, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  105 [10: JMP    ] PC: 10 DR:  2 SB:  0 RG: [9, 3, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  106 [ 1: CMP    ] PC:  1 DR:  2 SB:  0 RG: [9, 3, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  107 [ 2: JZ     ] PC:  2 DR:  2 SB:  0 RG: [9, 3, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  108 [ 3: MOD_RRV] PC:  3 DR:  2 SB:  0 RG: [9, 3, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  110 [ 4: JZ_R   ] PC:  4 DR:  2 SB:  0 RG: [9, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  112 [ 8: ADD_MMR] PC:  8 DR:  2 SB:  0 RG: [9, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  116 [ 9: INC    ] PC:  9 DR:  2 SB: 14 RG: [9, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  117 [10: JMP    ] PC: 10 DR:  2 SB: 14 RG: [10, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  118 [ 1: CMP    ] PC:  1 DR:  2 SB: 14 RG: [10, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  119 [ 2: JZ     ] PC:  2 DR:  2 SB: 14 RG: [10, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  120 [ 3: MOD_RRV] PC:  3 DR:  2 SB: 14 RG: [10, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  122 [ 4: JZ_R   ] PC:  4 DR:  2 SB:  0 RG: [10, 1, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  124 [ 5: MOD_RRV] PC:  5 DR:  2 SB:  0 RG: [10, 1, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  126 [ 6: JZ_R   ] PC:  6 DR:  2 SB:  0 RG: [10, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  128 [ 8: ADD_MMR] PC:  8 DR:  2 SB:  0 RG: [10, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  132 [ 9: INC    ] PC:  9 DR:  2 SB: 23 RG: [10, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  133 [10: JMP    ] PC: 10 DR:  2 SB: 23 RG: [11, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  134 [ 1: CMP    ] PC:  1 DR:  2 SB: 23 RG: [11, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  135 [ 2: JZ     ] PC:  2 DR:  2 SB: 23 RG: [11, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  136 [ 3: MOD_RRV] PC:  3 DR:  2 SB: 23 RG: [11, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  138 [ 4: JZ_R   ] PC:  4 DR:  2 SB:  0 RG: [11, 2, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  140 [ 5: MOD_RRV] PC:  5 DR:  2 SB:  0 RG: [11, 2, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  142 [ 6: JZ_R   ] PC:  6 DR:  2 SB:  0 RG: [11, 1, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  144 [ 7: JMP    ] PC:  7 DR:  2 SB:  0 RG: [11, 1, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  145 [ 9: INC    ] PC:  9 DR:  2 SB:  0 RG: [11, 1, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  146 [10: JMP    ] PC: 10 DR:  2 SB:  0 RG: [12, 1, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  147 [ 1: CMP    ] PC:  1 DR:  2 SB:  0 RG: [12, 1, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  148 [ 2: JZ     ] PC:  2 DR:  2 SB:  0 RG: [12, 1, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  149 [ 3: MOD_RRV] PC:  3 DR:  2 SB:  0 RG: [12, 1, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  151 [ 4: JZ_R   ] PC:  4 DR:  2 SB:  0 RG: [12, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  153 [ 8: ADD_MMR] PC:  8 DR:  2 SB:  0 RG: [12, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  157 [ 9: INC    ] PC:  9 DR:  2 SB: 33 RG: [12, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  158 [10: JMP    ] PC: 10 DR:  2 SB: 33 RG: [13, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  159 [ 1: CMP    ] PC:  1 DR:  2 SB: 33 RG: [13, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  160 [ 2: JZ     ] PC:  2 DR:  2 SB: 33 RG: [13, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  161 [ 3: MOD_RRV] PC:  3 DR:  2 SB: 33 RG: [13, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  163 [ 4: JZ_R   ] PC:  4 DR:  2 SB:  0 RG: [13, 1, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  165 [ 5: MOD_RRV] PC:  5 DR:  2 SB:  0 RG: [13, 1, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  167 [ 6: JZ_R   ] PC:  6 DR:  2 SB:  0 RG: [13, 3, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  169 [ 7: JMP    ] PC:  7 DR:  2 SB:  0 RG: [13, 3, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  170 [ 9: INC    ] PC:  9 DR:  2 SB:  0 RG: [13, 3, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  171 [10: JMP    ] PC: 10 DR:  2 SB:  0 RG: [14, 3, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  172 [ 1: CMP    ] PC:  1 DR:  2 SB:  0 RG: [14, 3, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  173 [ 2: JZ     ] PC:  2 DR:  2 SB:  0 RG: [14, 3, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  174 [ 3: MOD_RRV] PC:  3 DR:  2 SB:  0 RG: [14, 3, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  176 [ 4: JZ_R   ] PC:  4 DR:  2 SB:  0 RG: [14, 2, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  178 [ 5: MOD_RRV] PC:  5 DR:  2 SB:  0 RG: [14, 2, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  180 [ 6: JZ_R   ] PC:  6 DR:  2 SB:  0 RG: [14, 4, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  182 [ 7: JMP    ] PC:  7 DR:  2 SB:  0 RG: [14, 4, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  183 [ 9: INC    ] PC:  9 DR:  2 SB:  0 RG: [14, 4, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  184 [10: JMP    ] PC: 10 DR:  2 SB:  0 RG: [15, 4, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  185 [ 1: CMP    ] PC:  1 DR:  2 SB:  0 RG: [15, 4, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  186 [ 2: JZ     ] PC:  2 DR:  2 SB:  0 RG: [15, 4, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  187 [ 3: MOD_RRV] PC:  3 DR:  2 SB:  0 RG: [15, 4, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  189 [ 4: JZ_R   ] PC:  4 DR:  2 SB:  0 RG: [15, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  191 [ 8: ADD_MMR] PC:  8 DR:  2 SB:  0 RG: [15, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  195 [ 9: INC    ] PC:  9 DR:  2 SB: 45 RG: [15, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  196 [10: JMP    ] PC: 10 DR:  2 SB: 45 RG: [16, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  197 [ 1: CMP    ] PC:  1 DR:  2 SB: 45 RG: [16, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  198 [ 2: JZ     ] PC:  2 DR:  2 SB: 45 RG: [16, 0, 0, 0] 
  DEBUG    root:simulation.py:26 TICK:  199 [ 3: MOD_RRV] PC:  3 DR:  2 SB: 45 RG: [16, 0, 0, 0] 
  DEBUG    root:data_path.py:153 output: 233168 << 233168
  INFO     root:simulation.py:42 output_buffer: 233168
  INFO     root:simulation.py:68 End simulation
